src/main/java/org/elasticsearch/transport/netty/SizeHeaderFrameDecoder.java

```diff
@@ -68,6 +68,12 @@ protected Object decode(ChannelHandlerContext ctx, Channel channel, ChannelBuffe
         }
 
         int dataLen = buffer.getInt(buffer.readerIndex() + 2);
+        if (dataLen == NettyHeader.PING_DATA_SIZE) {
+            // discard the messages we read and continue, this is achieved by skipping the bytes
```

Comment:
we have an if check before in the if statement and send stream corrupt if not and I believe a test for it already